xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -55,69 +56,41 @@
     // The list virtual hosts that make up the route table.
     @Nullable
     final List<VirtualHost> virtualHosts;
-    // Listener contains the HttpFault filter.
-    final boolean hasFaultInjection;
-    @Nullable // Can be null even if hasFaultInjection is true.
-    final HttpFault httpFault;
+    // Filter instance names. Null if HttpFilter support is not enabled.
+    @Nullable final List<NamedFilter> filterChain;
     // Server side Listener.
     @Nullable
     final Listener listener;
 
     LdsUpdate(
-        long httpMaxStreamDurationNano, String rdsName, boolean hasFaultInjection,
-        @Nullable HttpFault httpFault) {
-      this(httpMaxStreamDurationNano, rdsName, null, hasFaultInjection, httpFault);
+        long httpMaxStreamDurationNano, String rdsName,
+        @Nullable List<NamedFilter> filterChain) {
+      this(httpMaxStreamDurationNano, rdsName, null, filterChain);
     }
 
     LdsUpdate(
         long httpMaxStreamDurationNano, List<VirtualHost> virtualHosts,
-        boolean hasFaultInjection, @Nullable HttpFault httpFault) {
-      this(httpMaxStreamDurationNano, null, virtualHosts, hasFaultInjection, httpFault);
+        @Nullable List<NamedFilter> filterChain) {
+      this(httpMaxStreamDurationNano, null, virtualHosts, filterChain);
     }
 
     private LdsUpdate(
         long httpMaxStreamDurationNano, @Nullable String rdsName,
-        @Nullable List<VirtualHost> virtualHosts, boolean hasFaultInjection,
-        @Nullable HttpFault httpFault) {
+        @Nullable List<VirtualHost> virtualHosts, @Nullable List<NamedFilter> filterChain) {
       this.httpMaxStreamDurationNano = httpMaxStreamDurationNano;
       this.rdsName = rdsName;
       this.virtualHosts = virtualHosts == null
           ? null : Collections.unmodifiableList(new ArrayList<>(virtualHosts));
-      this.hasFaultInjection = hasFaultInjection;
-      this.httpFault = httpFault;
+      this.filterChain = filterChain == null ? null : Collections.unmodifiableList(filterChain);
       this.listener = null;
     }
 
     LdsUpdate(Listener listener) {
       this.listener = listener;
       this.httpMaxStreamDurationNano = 0L;
       this.rdsName = null;
+      this.filterChain = null;
       this.virtualHosts = null;
-      this.hasFaultInjection = false;
-      this.httpFault = null;
-    }
-
-    @Override
-    public int hashCode() {
```

Comment:
Why this is deleted? You cannot delete them, XdsClient replies on equality to determine if the xDS response contains duplicated update.

Reply:
My bad, deleted by accident.